plugin/pkg/namespacedcloudprofile/validator/admission.go

```diff
@@ -0,0 +1,184 @@
+	gardencorev1beta1listers "github.com/gardener/gardener/pkg/client/core/listers/core/v1beta1"
+	plugin "github.com/gardener/gardener/plugin/pkg"
+)
+
+// Register registers a plugin.
+func Register(plugins *admission.Plugins) {
+	plugins.Register(plugin.PluginNameNamespacedCloudProfileValidator, func(_ io.Reader) (admission.Interface, error) {
+		return New()
+	})
+}
+
+// ValidateNamespacedCloudProfile contains listers and admission handler.
+type ValidateNamespacedCloudProfile struct {
+	*admission.Handler
+	cloudProfileLister gardencorev1beta1listers.CloudProfileLister
+	readyFunc          admission.ReadyFunc
+}
+
+var (
+	_          = admissioninitializer.WantsCoreInformerFactory(&ValidateNamespacedCloudProfile{})
+	readyFuncs []admission.ReadyFunc
+)
+
+// New creates a new ValidateNamespacedCloudProfile admission plugin.
+func New() (*ValidateNamespacedCloudProfile, error) {
+	return &ValidateNamespacedCloudProfile{
+		Handler: admission.NewHandler(admission.Create, admission.Update),
+	}, nil
+}
+
+// AssignReadyFunc assigns the ready function to the admission handler.
+func (v *ValidateNamespacedCloudProfile) AssignReadyFunc(f admission.ReadyFunc) {
+	v.readyFunc = f
+	v.SetReadyFunc(f)
+}
+
+// SetCoreInformerFactory gets Lister from SharedInformerFactory.
+func (v *ValidateNamespacedCloudProfile) SetCoreInformerFactory(f gardencoreinformers.SharedInformerFactory) {
+	cloudProfileInformer := f.Core().V1beta1().CloudProfiles()
+	v.cloudProfileLister = cloudProfileInformer.Lister()
+
+	readyFuncs = append(readyFuncs, cloudProfileInformer.Informer().HasSynced)
+}
+
+// ValidateInitialization checks whether the plugin was correctly initialized.
+func (v *ValidateNamespacedCloudProfile) ValidateInitialization() error {
+	if v.cloudProfileLister == nil {
+		return errors.New("missing cloudProfile lister")
+	}
+	return nil
+}
+
+var _ admission.ValidationInterface = &ValidateNamespacedCloudProfile{}
+
+// Validate validates the NamespacedCloudProfile
+func (v *ValidateNamespacedCloudProfile) Validate(_ context.Context, a admission.Attributes, _ admission.ObjectInterfaces) error {
+	// Wait until the caches have been synced
+	if v.readyFunc == nil {
+		v.AssignReadyFunc(func() bool {
+			for _, readyFunc := range readyFuncs {
+				if !readyFunc() {
+					return false
+				}
+			}
+			return true
+		})
+	}
+	if !v.WaitForReady() {
+		return admission.NewForbidden(a, errors.New("not yet ready to handle request"))
+	}
+
+	if a.GetKind().GroupKind() != gardencorev1beta1.Kind("NamespacedCloudProfile") {
+		return nil
+	}
+
+	if a.GetSubresource() != "" {
+		return nil
+	}
+
+	var oldNamespacedCloudProfile = &gardencorev1beta1.NamespacedCloudProfile{}
+
+	namespacedCloudProfile, convertIsSuccessful := a.GetObject().(*gardencorev1beta1.NamespacedCloudProfile)
+	if !convertIsSuccessful {
+		return apierrors.NewInternalError(errors.New("could not convert object to NamespacedCloudProfile"))
+	}
+
+	// Exit early if shoot spec hasn't changed
```

Comment:
Copy paste error?

Reply:
Unfortunately yes, fixed in https://github.com/gardener/gardener/pull/9440/commits/bc2bd226df5058cdde6c446ccabd89b024c97d49.